test/integration/targets/ec2_instance/tasks/iam_instance_role.yml

```diff
@@ -0,0 +1,62 @@
+- name: set connection information for all tasks
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ aws_region }}"
+  no_log: true
+
+- block:
+    - name: Create IAM role for test
+      iam_role:
+        name: "{{ resource_prefix }}-test-policy"
+        assume_role_policy_document: "{{ lookup('file','assume-role-policy.json') }}"
+        state: present
+        create_instance_profile: yes
+        managed_policy:
+        - AmazonEC2ContainerServiceRole
+        <<: *aws_connection_info
+      register: iam_role
+
+    - name: Wait for IAM role to be available, otherwise the next step will fail (Invalid IAM Instance Profile name)
+      command: sleep 10
```

Comment:
Can you use iam_role_facts instead of the sleep to ensure the role exists? You could try until iam_roles in the results is truthy. Besides that, looks great to me.

Reply:
Sure! Didn't thought about that :)

Reply:
So, I've tried to use the `iam_role_facts` before making the `ec2_instance` call. 
And it doesn't help in this case, because `iam_role_facts` returns right away with the correct values (I've even updated the modules to also fetch information for instance profiles, and it is correct)

My guess is that even if it's visible in IAM, it's not visible right away in ec2.

I'm not really sure if there is a better way to solve this, than the `sleep` command. I've looked at the ec2 boto methods, and there is none that checks the existence of an instance_profile

Reply:
Ah, I see. Possible solution would be to make a customer waiter for iam_role, in that case (https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/aws/waiters.py) but this looks good to me as it is.